website/src/pages/api/leaderboard.ts

```diff
@@ -5,6 +5,12 @@ import { getBackendUserCore } from "src/lib/users";
 import { LeaderboardTimeFrame } from "src/types/Leaderboard";
 
 export default withoutRole("banned", async (req, res, token) => {
+  if (!token.backendUserId) {
+    // user has not yet accepted the terms of service, and therefor does not yet exist in the backend
+    /// skip the request entirely
+    return res.status(200).json(null);
```

Comment:
Does this break the UI?

Reply:
No it should not, the leaderboard will be empty.

It is also not visble since the ToS wrapper covers the entire screen, after the user accepts ToS the webpage is refreshed and a new request is made.